test/FastTests/Corax/StreamingOptimization_QueryBuilder.cs

```diff
@@ -551,20 +553,30 @@ public EnvTest(ITestOutputHelper output) : base(output)
         //So let's add it directly ;-) When something will change about data structure remember to update this as well.
         public void Init(IndexFieldsMapping mapping, bool hasMultipleValues)
         {
-            if (hasMultipleValues == false)
-                return;
-
             TransactionPersistentContext transactionPersistentContext = new(true);
             using (var transaction = Env.WriteTransaction(transactionPersistentContext))
             {
-                var indexMetadata = transaction.CreateTree(Constants.IndexMetadataSlice);
-                foreach (var binding in mapping)
+                using (var indexWriter = new IndexWriter(transaction, mapping))
                 {
-                    Tree tree = transaction.CreateTree(Constants.IndexWriter.MultipleTermsInField);
+                    const string someValue = "3.14";
                     
-                    tree.Add(binding.FieldName, 1);
-                    tree.Add(binding.FieldNameDouble, 1);
-                    tree.Add(binding.FieldNameLong, 1);
+                    using (var builder = indexWriter.Index("entryKey"))
+                    {
+                        foreach (var field in mapping)
+                        {
+                            if (hasMultipleValues)
+                                builder.IncrementList();
+
+                            builder.Write(field.FieldId, Encoding.UTF8.GetBytes(someValue), 3, 3.14D);
+                            
+                            if (hasMultipleValues)
+                            {
+                                builder.Write(field.FieldId, Encoding.UTF8.GetBytes(someValue), 3, 3.14D);
+                                builder.DecrementList();
+                            }
+                        }
+                    }
+                    indexWriter.Commit();
```

Comment:
Why did you modify this test?

Reply:
We're only testing query building here, and previously we didn't insert any index terms because they weren't needed. Now, because we ignore ordering by fields that have no terms when building query, we need to insert some data.